libs/base/Data/List/Elem.idr

```diff
@@ -19,11 +19,11 @@ data Elem : a -> List a -> Type where
      There : Elem x xs -> Elem x (y :: xs)
 
 export
-Uninhabited (Here = There e) where
+Uninhabited (Here {x} {xs} = There {x = x'} {y} {xs} e) where
```

Comment:
If you're going fully heterogeneous, why not have two different `xs` on each sides?

Reply:
good point. `xs` are now different

Reply:
Since `xs` are now different maybe we should add something like that? 
`Either (Not (x = x')) (Not xs = xs') => Uninhabited (Here {x} {xs} = Here {x = x'} {xs = xs'}) where`
Basically, if `x` or `xs` are different, `Here` and `There` will be different as well.

Reply:
> Either (Not (x = x')) (Not xs = xs') => Uninhabited (Here {x} {xs} = Here {x = x'} {xs = xs'})

I think this can be proven for any `Biinjective`, not only for this particular one